app/src/main/java/org/glucosio/android/tools/GlucosioAlarmManager.java

```diff
@@ -0,0 +1,65 @@
+package org.glucosio.android.tools;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+
+import org.glucosio.android.db.DatabaseHandler;
+import org.glucosio.android.db.Reminder;
+import org.glucosio.android.receivers.GlucosioBroadcastReceiver;
+
+import java.util.Calendar;
+import java.util.List;
+
+public class GlucosioAlarmManager {
+    List<Reminder> reminders;
+    Context context;
+    AlarmManager alarmMgr;
+    DatabaseHandler db;
+
+    public GlucosioAlarmManager(Context context) {
+        this.context = context;
+        this.db = new DatabaseHandler(context);
+        this.alarmMgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+    }
+
+    public void setAlarms() {
+        reminders = db.getReminders();
+
+        // if there are some active alarms, start receiver to set them at boot
+        enableBootReceiver(db.areRemindersActive());
```

Comment:
Realm instances cannot be passed across threads. Other developers said that Realm is actually working on that and "Realm's performance is surprising and even running queries on the main thread isn't that much of an issue". Anyway, I've fixed this particular method and made a db request just one time.